plugins/synaptics-mst/fu-synaptics-mst-device.c

```diff
@@ -79,15 +79,17 @@ fu_synaptics_mst_device_finalize(GObject *object)
 static void
 fu_synaptics_mst_device_init(FuSynapticsMstDevice *self)
 {
+	FuUdevDeviceFlags flags =
+	    FU_UDEV_DEVICE_FLAG_OPEN_READ | FU_UDEV_DEVICE_FLAG_VENDOR_FROM_PARENT;
+	if (fu_udev_device_get_dev(FU_UDEV_DEVICE(self)) != NULL)
+		flags |= FU_UDEV_DEVICE_FLAG_OPEN_WRITE;
```

Comment:
Is this safe? Do we ever set the udev dev after the mst device has been set up?

Reply:
We shouldn't be at least..  Synaptics MST device inherits from a udev device, and that's the thing we're setting flags on